pkg/operator/certrotation/signer_test.go

```diff
@@ -19,6 +26,170 @@ import (
 	"github.com/openshift/library-go/pkg/operator/events"
 )
 
+func TestRotatedSigningCASecretWithMultipleControllers(t *testing.T) {
+	ns, name := "ns", "test-signer"
+	// represents a secret that was created before 4.7 and
+	// hasn't been updated until now (upgrade to 4.15)
+	existing := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:       ns,
+			Name:            name,
+			ResourceVersion: "10",
+			Annotations: map[string]string{
+				"auth.openshift.io/certificate-not-after":  "2108-09-08T22:47:31-07:00",
+				"auth.openshift.io/certificate-not-before": "2108-09-08T20:47:31-07:00",
+			},
+		},
+		Type: "SecretTypeTLS",
+		Data: map[string][]byte{"tls.crt": {}, "tls.key": {}},
+	}
+	if err := setSigningCertKeyPairSecret(existing, 24*time.Hour); err != nil {
+		t.Fatal(err)
+	}
+
+	// give it a second so we have a unique signer name,
```

Comment:
Can we change the way signer names are chosen so that they are practically guaranteed to be unique? We were looking at kube-apiserver logs last week and observed that the issuer name was the same for the external and internal serving certs. Given the issue context, it's not unrealistic to imagine two successive rotations producing signers with the same name.

Reply:
It uses `time.Now().Unix()`, we could use `UnixMicro` or `UnixNano` to significantly reduce the chances of issuer name being same. 
FWIW, i think it is outside the scope of this PR, i don't think that changing it will have a bad ripple effect on any cluster, not sure any tool relies on the current format